lib/dal/src/action.rs

```diff
@@ -231,4 +304,74 @@ impl Action {
 
         Ok(result)
     }
+
+    pub async fn func_execution(&self, ctx: &DalContext) -> ActionResult<Option<FuncExecution>> {
+        Ok(match self.func_execution_pk {
+            Some(pk) => Some(FuncExecution::get_by_pk(ctx, &pk).await?),
+            None => None,
+        })
+    }
+
+    pub async fn run(
+        ctx: &DalContext,
+        id: ActionId,
+    ) -> ActionResult<Option<DeprecatedActionRunResult>> {
+        let component_id = Action::component_id(ctx, id)
+            .await?
+            .ok_or(ActionError::ComponentNotFoundForAction(id))?;
+
+        let prototype_id = Action::prototype_id(ctx, id)
+            .await?
+            .ok_or(ActionError::PrototypeNotFoundForAction(id))?;
+        let prototype = ActionPrototype::get_by_id(ctx, prototype_id).await?;
+
+        Action::set_state(ctx, id, ActionState::Running).await?;
+
```

Comment:
I think we want to commit here so that this status is rebased on head

Reply:
That's a good point, I may move this to the job definition to make it less confusing, as commiting inside of the regular dal methods feels weird.

Good catch!

Reply:
Yup, definitely want to commit in this phase of running, and completely agree with Paulo that this probably belongs in the job definition.